### DISCUSSION TITLE
uniform response format

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#38] [Q&A]
Hi, @wu-clan 

Ideally, any errors return a uniform format:
```
{
    "code": 404,
    "msg": "Not Found",
    "data": null
}
```

But, if no parameters are passed during login, an error will be reported:

```
{
    "detail": [{
        "loc": ["body", "username"],
        "msg": "field required",
        "type": "value_error.missing"
    }, {
        "loc": ["body", "password"],
        "msg": "field required",
        "type": "value_error.missing"
    }]
}
```
It doesn't seem to go through all_exception_handler, pydantic custom errors are too cumbersome

The interface is originally a get method, but uses the post method
```
{
    "detail": "Method Not Allowed"
}
```

**Comment by wu-clan** [comment]
Hi, @downdawn

I checked the problem, I put the data validation exception handling in the global exception detection and based on the jump, I found that they both start at Exception and I thought they would work, but it seems to be nested deeper and not detecting the source of the pydantic exception

Solution, separate data validation exception handling and put it before Exception

**Comment by wu-clan** [comment]
Also made some changes to the login interface return content

**Reply by downdawn** [reply]
Nice
